code/core/arm9/source/GbaColorCorrection/ColorLut.cpp

```diff
@@ -0,0 +1,101 @@
+#include "common.h"
+#include <algorithm>
+#include "ColorLut.h"
+#include "GammaLut.h"
+
+// Based on https://gist.github.com/profi200/bfa7be60b3eecb8c43f59000f626c743
+
+u16 gColorLut[COLOR_LUT_SIZE];
+
+// When the 2d engine converts from 5 to 6 bit, the lsb bit will always be zero (i.e. 31 -> 62)
+static constexpr u32 rgb8ToRgb5(u32 value8)
+{
+    u32 value5 = (value8 * 63 + 255) / (255 * 2);
+    if (value5 > 31)
+        return 31;
+    return value5;
+}
+
+// Convert RGB8 to RGB6 (for the 6-bit green)
+static constexpr u32 rgb8ToRgb6(u32 value8)
+{
+    return (value8 * 63 + 128) / 255;
+}
+
+// Apply correction matrix from selected color profile
+static void applyColorMatrix(const fix32<12> matrix[3][3],
+                             fix32<12> r, fix32<12> g, fix32<12> b,
+                             fix32<12>& outR, fix32<12>& outG, fix32<12>& outB)
+{
+    fix32<12> newR = (matrix[0][0] * r + matrix[0][1] * g + matrix[0][2] * b);
+    fix32<12> newG = (matrix[1][0] * r + matrix[1][1] * g + matrix[1][2] * b);
+    fix32<12> newB = (matrix[2][0] * r + matrix[2][1] * g + matrix[2][2] * b);
+
+    // We need to clamp specifically at this step or else color will get ruined
+    outR = std::clamp((newR.Int()), 0, 255);
+    outG = std::clamp((newG.Int()), 0, 255);
+    outB = std::clamp((newB.Int()), 0, 255);
```

Comment:
```suggestion
    outR = std::clamp(newR.Int(), 0, 255);
    outG = std::clamp(newG.Int(), 0, 255);
    outB = std::clamp(newB.Int(), 0, 255);
```
I assume these extra parenthesis are unnecessary